app/app.go

```diff
@@ -432,7 +427,7 @@ func NewSifApp(
 		vesting.NewAppModule(app.AccountKeeper, app.BankKeeper),
 		bank.NewAppModule(appCodec, app.BankKeeper, app.AccountKeeper),
 		capability.NewAppModule(appCodec, *app.CapabilityKeeper),
-		crisis.NewAppModule(&app.CrisisKeeper, skipGenesisInvariants),
+		crisis.NewAppModule(&app.CrisisKeeper, true),
```

Comment:
Invariants are certain conditions which are checked , which should hold true to certify the network is not byzantine . 

The invariants are asserted in two places 
- End of every block 
https://github.com/cosmos/cosmos-sdk/blob/d3a8e1e9535e4d980723fbabdc274cd55cf965ae/x/crisis/abci.go#L20

- InitGenesis of the Crisis module . 
https://github.com/cosmos/cosmos-sdk/blob/d3a8e1e9535e4d980723fbabdc274cd55cf965ae/x/crisis/module.go#L147

I noticed during local testing , the network crash on upgrade is from the invariant checks which happen during init genesis . And these crashes are sporadic . 
![image](https://user-images.githubusercontent.com/72057696/145675691-faf6299c-71aa-4ca5-b72f-e48fb4b5be00.png)
Attached image show a sequence in which the invariant checks succed on initgenesis ,but it does not run successfully always 

The fix in this PR , diables the invarient check on init genesis and turns on these checks to happen at the end of each block . 
In essence in a node is malicious  , and the upgrade happens at height 6 , it will only crash at height 7 

The invariant checks will eat some compute , but we will need to test that on a cloud env

Reply:
also the two values can be set through sifapp initialization , if thats required

Reply:
@Jedi2002 that's a good idea to set these values through app init, also it would be nice to comment in code as well to make sure that it is 100% clear why we make this change